src/Kestrel.Core/Http2Limits.cs

```diff
@@ -14,11 +14,12 @@ public class Http2Limits
         private int _maxStreamsPerConnection = 100;
         private int _headerTableSize = (int)Http2PeerSettings.DefaultHeaderTableSize;
         private int _maxFrameSize = (int)Http2PeerSettings.DefaultMaxFrameSize;
+        private int _maxRequestHeaderFieldSize = 4096;
```

Comment:
It should be at least 8kb so it can handle the same request path & query values as HTTP/1